sql-plugin/src/main/java/com/nvidia/spark/rapids/SpillableKudoTable.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids;
+
+import ai.rapids.cudf.HostMemoryBuffer;
+import com.nvidia.spark.rapids.jni.kudo.KudoTable;
+import com.nvidia.spark.rapids.jni.kudo.KudoTableHeader;
+
+public class SpillableKudoTable extends KudoTable {
```

Comment:
if this was in scala, it might make more sense, as it's only used from scala.

Reply:
For me I would prefer to have SpillableKudoTable not extend KudoTable.

Conceptually I want to say that I have a list of `SpillableKudoTable`s, and then when I want to concat and serialize them into a `Table` I convert all of them into regular `KudoTable`s which guarantees that they are all resident in memory, do the concat operation, and then close them when I am done. 


This implies that a `SpillableKudoTable` is a `KudoTable`, which we can make it work, but as @abellina pointed out it makes the code much more complicated. If you have `SpillableKudoTable` produce a `KudoTable`, then we don't need `guaranteeSpillable`. The act of getting the KudoTable out implicitly increments the reference count and make it not spillable until it goes out of scope.